pkg/combustion/registry.go

```diff
@@ -173,6 +169,28 @@ func createRegistryCommand(ctx *image.Context, commandName string, args []string
 	return cmd, logFile, nil
 }
 
-func IsEmbeddedArtifactRegistryEmpty(registry image.EmbeddedArtifactRegistry) bool {
-	return len(registry.HelmCharts) == 0 && len(registry.ContainerImages) == 0
+func IsEmbeddedArtifactRegistryAndKubernetesManifestsEmpty(ctx *image.Context) bool {
+	return len(ctx.ImageDefinition.EmbeddedArtifactRegistry.ContainerImages) == 0 && len(ctx.ImageDefinition.Kubernetes.HelmCharts) == 0
+}
+
+func writeHaulerManifestAndGenerateTar(ctx *image.Context, images []image.ContainerImage, charts []image.HelmChart, registryOrigin string) error {
```

Comment:
On my previous note, once you start having to call functions `doThisAndThis`, they're no longer easy to understand, maintain and test. What we can do here is to split the flows and have `configureContainerImages` and `configureHelmCharts` functions or something similar. Each one would be working with its separate template and would be called based on its own condition e.g. 
```
if len(ctx.ImageDefinition.EmbeddedArtifactRegistry.ContainerImages) != 0 {
   configureContainerImages()
}
```

This will allow better testing and easier upgrades once we get to adding manifests in as well.